src/preamble.js

```diff
@@ -780,10 +780,23 @@ var memoryInitializer = null;
 #include "memoryprofiler.js"
 #endif
 
-#if PTHREAD_POOL_SIZE > 0 && PTHREAD_POOL_DELAY_LOAD != 1
-// To work around https://bugzilla.mozilla.org/show_bug.cgi?id=1049079, warm up a worker pool before starting up the application.
-if (!ENVIRONMENT_IS_PTHREAD) addOnPreRun(function() { if (typeof SharedArrayBuffer !== 'undefined') { addRunDependency('pthreads'); PThread.allocateUnusedWorkers({{{PTHREAD_POOL_SIZE}}}, function() { removeRunDependency('pthreads'); }); }});
+#if USE_PTHREADS && PTHREAD_POOL_SIZE
+
+var pthreadPoolSize = Module['pthreadPoolSize'] || {{{ PTHREAD_POOL_SIZE }}};
+
+#if PTHREAD_POOL_DELAY_LOAD != 1
+if (!ENVIRONMENT_IS_PTHREAD && pthreadPoolSize > 0) {
+  addOnPreRun(function() {
+    if (typeof SharedArrayBuffer !== 'undefined') {
+      addRunDependency('pthreads');
+      PThread.allocateUnusedWorkers(pthreadPoolSize, function() {
```

Comment:
Uh hmm we have now both `PThread.allocateUnusedWorkers` and `PThread.createNewWorkers(pthreadPoolSize);`, that seems very wrong.  To me this looks like we are allocating 2x the amount of Workers that are needed? Or perhaps I am very confused here. The intent of #9394 where this came into existence does not make sense to me. hmm...

Reply:
I believe the idea is that creating and instantiating the workers is a separate thing. One can create the Worker objects without blocking startup on being able to instantiate them with the info they require etc. But we can discuss in your comments over there.

However, all of that is separate from this PR, and should not block it, unless I'm missing something?

Reply:
Perhaps this PR can be relayered on top of https://github.com/emscripten-core/emscripten/pull/10269/files#diff-db41bea94577c2dd9b0eef0308b06cf9R36 ?

Reply:
Maybe we can relayer on top of that, but that one seems to need discussion (so we figure out the history there) so unclear how long it will take to land, while this one is ready to go, isn't it? :smile: Or is there some reason you'd prefer to delay landing this?